app/models/ems_refresh.rb

```diff
@@ -178,16 +178,18 @@ def self.queue_merge(targets, ems, create_task = false)
                 end
 
       item.merge(
-        :args         => [targets],
-        :task_id      => task_id,
-        :msg_timeout  => queue_timeout,
-        :miq_callback => {
+        :args        => [targets],
+        :task_id     => task_id,
+        :msg_timeout => queue_timeout
+      )
+      unless task_id.nil?
+        item[:miq_callback] = {
           :class_name  => 'MiqTask',
           :method_name => :queue_callback,
           :instance_id => task_id,
           :args        => ['Finished']
         }
-      )
+      end
     end
```

Comment:
Need to yield `item` here since `MiqQueue` is using the return of the block [here](https://github.com/ManageIQ/manageiq/blob/d4e12c902e87452a014b5b0fb99863a50ddccaca/app/models/miq_queue.rb#L2400